lib/elixir/lib/module.ex

```diff
@@ -1824,22 +1802,11 @@ defmodule Module do
   defp args_count([], total, defaults), do: {total, defaults}
 
   @doc false
-  def check_derive_behaviours_and_impls(env, set, bag, all_definitions) do
+  def check_derive_behaviours_and_impls(env, set, bag) do
     check_derive(env, set, bag)
     behaviours = bag_lookup_element(bag, {:accumulate, :behaviour}, 2)
-    impls = bag_lookup_element(bag, :impls, 2)
-    callbacks = check_behaviours(env, behaviours)
-
-    pending_callbacks =
-      if impls != [] do
-        {non_implemented_callbacks, contexts} = check_impls(env, behaviours, callbacks, impls)
-        warn_missing_impls(env, non_implemented_callbacks, contexts, all_definitions)
-        non_implemented_callbacks
-      else
-        callbacks
-      end
+    Module.Behaviour.force_runtime_dependencies(behaviours, env)
```

Comment:
I would move only the logic of this particular function in particular here. Everything else in Module.Behaviour is 💯 .

Reply:
Yeah, I hesitated. Will move here 👍